sdk/python/feast/infra/online_stores/datastore.py

```diff
@@ -0,0 +1,257 @@
+# Copyright 2021 The Feast Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import itertools
+from datetime import datetime
+from multiprocessing.pool import ThreadPool
+from typing import Any, Callable, Dict, Iterator, List, Optional, Sequence, Tuple, Union
+
+import mmh3
+
+from feast import Entity, FeatureTable, utils
+from feast.feature_view import FeatureView
+from feast.infra.key_encoding_utils import serialize_entity_key
+from feast.infra.online_stores.online_store import OnlineStore
+from feast.protos.feast.types.EntityKey_pb2 import EntityKey as EntityKeyProto
+from feast.protos.feast.types.Value_pb2 import Value as ValueProto
+from feast.repo_config import DatastoreOnlineStoreConfig, RepoConfig
+
+try:
+    from google.auth.exceptions import DefaultCredentialsError
+    from google.cloud import datastore
+except ImportError as e:
+    from feast.errors import FeastExtrasDependencyImportError, FeastProviderLoginError
+
+    raise FeastExtrasDependencyImportError("gcp", str(e))
+
+
+ProtoBatch = Sequence[
+    Tuple[EntityKeyProto, Dict[str, ValueProto], datetime, Optional[datetime]]
+]
+
+
+class DatastoreOnlineStore(OnlineStore):
+    """
+    OnlineStore is an object used for all interaction between Feast and the service used for offline storage of
+    features.
+    """
+
+    @classmethod
+    def setup(
+        cls,
```

Comment:
Just out of curiosity, is there a reason we went for @classmethods here as opposed to normal classes (and using self) or static methods?

Reply:
Not really any reason, I was thinking of making this regular instance methods too